rig/machine_control/machine_controller.py

```diff
@@ -1381,6 +1395,48 @@ def __init__(self, machine_controller, x, y, start_address, end_address):
         # Current offset from start address
         self._offset = 0
 
+    def __getitem__(self, sl):
+        """Get a new file-like view of SDRAM covering the range indicated by
+        the slice.
+
+        For example, if `f` is a `MemoryIO` covering a 100 byte region of SDRAM
+        then::
+
+            g = f[0:10]
+
+        Creates a new `MemoryIO` referring to just the first 10 bytes of `f`.
+
+        Raises
+        ------
+        ValueError
+            If the slice is not contiguous.
+        """
+        if isinstance(sl, slice) and (sl.step is None or sl.step == 1):
+            # Get the start and end addresses
+            if sl.start is None:
+                start_address = self._start_address
+            elif sl.start < 0:
+                start_address = max(0, self._end_address + sl.start)
```

Comment:
The above may allow a start_address which is < the original start address (i.e. to grow the range of memory the MemoryIO can access). This would be a bug in my eyes.